### PR TITLE
cleanup & compatibility fixes

### PR DESCRIPTION
**Pull request by iffa**
- babel core, types in devDependencies
- import directly from babel types
- don't include src/ in npm package

reason for these changes:

``` bash
# with current @latest version:
error - ./node_modules/next/dist/client/next-dev.js
Error: Cannot find module '@babel/core'
Require stack:
- node_modules\babel-plugin-superjson-next\dist\index.js

# with changes from this PR:
ready - started server on http://localhost:3000
info  - Using external babel configuration from .babelrc
event - compiled successfully
```

can be reproduced with the basic project setup from create-next-app.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "compilerOptions": {
     "declaration": true,
     "sourceMap": true,
+    "lib": ["es2016"],
```
Comment (Skn0tt) [inline tsconfig.json:6]: Hi @iffa! Thanks for contributing this PR, it LGTM. Could you explain why this change above is needed? I don't see any ES2016-related changes in the rest of the PR.

Comment (iffa) [reply tsconfig.json:6]: IIRC this was only done because WebStorm was not particularly happy about the code using ES2016 functionality (i.e. Array.prototype.includes) without explicitly specifying that it is used

**Comment by Skn0tt** [comment]
I'm surprised that this resolves `Error: Cannot find module '@babel/core'`. Adding `@babel/core` to `devDependencies` shouldn't change anything since `devDependencies` aren't downloaded, anyway - what am I missing?

**Comment by iffa** [comment]
The reason this resolves the issue is due to the import declaration changes I made (types not imported from `@babel/core` but directly). The `devDependencies`-additions are probably unnecessary but I added them to be explicit. I can push some changes to the PR if you'd like.

**Comment by Skn0tt** [comment]
Thanks for explaining! That makes sense ☺️ Don't worry, I'll make the last few necessary edits myself :D

**Comment by Skn0tt** [comment]
@allcontributors add iffa for code
